sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -314,7 +338,9 @@ private[joins] object UnsafeHashedRelation {
       key: Seq[Expression],
       sizeEstimate: Int,
       taskMemoryManager: TaskMemoryManager,
-      isNullAware: Boolean = false): HashedRelation = {
+      isNullAware: Boolean = false,
```

Comment:
If isLookupAware is not implemented to work with isNullAware then we should assert that the two are mutually exclusively enabled.

Reply:
@agrawaldevesh - updated with a check to throw exception in this case.